### ISSUE TITLE
[Feature][CDC] Support oceanbase cdc connector source

### ISSUE THREAD

**Issue by hailin0** [apache/seatunnel#7198]
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Currently, we don't support reading oceanbase change data capture (CDC), does anyone want to add support for oceanbase?

docs
https://github.com/oceanbase/oblogclient

reference
https://github.com/apache/seatunnel/tree/dev/seatunnel-connectors-v2/connector-cdc/connector-cdc-mysql
https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e
https://github.com/apache/seatunnel/tree/dev/docs

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by YOMO-Lee** [comment]
please assign to me ,thx

**Comment by juliansysu163** [comment]
Is it in the works?

**Comment by YOMO-Lee** [comment]
> Is it in the works?

Yes, but I have encountered some problems and don't know how to solve them

**Comment by hailin0** [comment]
Hi @YOMO-Lee 
Is there any progress on this task?

**Comment by hawk9821** [comment]
Please assigned to me, ths .

**Comment by YOMO-Lee** [comment]
这是来自QQ邮箱的自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by davidzollo** [comment]
> Please assigned to me, ths .

@hawk9821 , How is it going on?  you can refer the implemention in `Flink CDC`

**Comment by YOMO-Lee** [comment]
这是来自QQ邮箱的自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by YOMO-Lee** [comment]
这是来自QQ邮箱的自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by hawk9821** [comment]
> > Please assigned to me, ths .
> 
> [@hawk9821](https://github.com/hawk9821) , How is it going on? you can refer the implemention in `Flink CDC`

**Comment by davidzollo** [comment]
This issue is still a good place to track the OceanBase CDC source connector, but it needs a more implementation-focused scope so contributors can claim it directly.

## Proposed scope
Add a dedicated `connector-cdc-oceanbase` source connector under `seatunnel-connectors-v2/connector-cdc`.

## Functional expectations
- Read both snapshot data and incremental change events from OceanBase.
- Support multi-table capture through SeaTunnel's CDC multi-table model.
- Integrate with SeaTunnel's checkpoint/recovery flow and preserve restart correctness.
- Surface CDC rows in the existing SeaTunnel row format so downstream multi-table sinks and schema evolution paths can consume them.
- Provide English and Chinese connector docs.
- Add focused tests, including at least one runnable integration or E2E path.

## Suggested acceptance criteria
- The connector can complete an initial snapshot and continue consuming incremental changes.
- Restart from checkpoint does not re-read or silently skip committed change events.
- At least one multi-table path is validated.
- Connector docs describe required privileges, startup configuration, and known limitations.

## Out of scope for the first delivery
- Sink connector work.
- Cross-connector refactors unrelated to OceanBase.
- New CDC metadata or routing DSL changes.

**Comment by YOMO-Lee** [comment]
这是来自QQ邮箱的自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by davidzollo** [comment]
Superseded by #11049, which provides a new claimable implementation scope, explicit first-delivery boundary, suggested module touch points, and acceptance criteria for an OceanBase CDC source connector.
